### PR TITLE
Filter check, get credentials from token

### PR DESCRIPTION
**Pull request by 0ffffh**

### CODE REVIEW

File: goods-partner-backend/src/main/java/com/goodspartner/service/security/impl/DefaultJwtService.java [hunk 1]
```diff
@@ -82,11 +86,25 @@ public Optional<UsernamePasswordAuthenticationToken> authenticateRefreshToken(St
...
-        User user = userService.findByLogin(login);// TODO Try to remove fetching from DB
-        GPUserDetails gpUserDetails = GPUserDetails.fromUser(user);
+        String username = claims.get(USERNAME_CLAIM, String.class);
+        List<GrantedAuthority> grantedAuthorities = getGrantedAuthorities(claims);
+
+        GPUserDetails gpUserDetails = GPUserDetails.builder()
+                .login(login)
+                .username(username)
+                .authorities(grantedAuthorities)
+                .build();
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/security/impl/DefaultJwtService.java:96]: isActive = false?

File: goods-partner-backend/src/main/java/com/goodspartner/service/security/impl/DefaultJwtService.java [hunk 2]
```diff
@@ -63,6 +65,8 @@ private String createToken(GPUserDetails user, SecretKey secret, long expiration
         String login = user.getLogin();
         return Jwts.builder()
                 .setSubject(login)
+                .claim(AUTHORITY_CLAIM, user.getAuthorities())
+                .claim(USERNAME_CLAIM, user.getUsername())
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/security/impl/DefaultJwtService.java:69]: make sense

File: goods-partner-backend/src/main/java/com/goodspartner/service/security/impl/DefaultJwtService.java (hunk 1 above)
Comment (belosh59) [reply goods-partner-backend/src/main/java/com/goodspartner/service/security/impl/DefaultJwtService.java:96]: in this case

File: goods-partner-backend/src/main/java/com/goodspartner/service/security/impl/DefaultJwtService.java [hunk 3]
```diff
@@ -63,6 +65,8 @@ private String createToken(GPUserDetails user, SecretKey secret, long expiration
         String login = user.getLogin();
         return Jwts.builder()
                 .setSubject(login)
```
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/security/impl/DefaultJwtService.java:67]: but why do we need subject at all, lets put everything under the claims

Comment (0ffffh) [reply goods-partner-backend/src/main/java/com/goodspartner/service/security/impl/DefaultJwtService.java:67]: sure we can, but
claims are pieces of information asserted about a subject
https://www.rfc-editor.org/rfc/rfc7519#section-4.1.2
https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-token-claims

**Review by belosh59 — APPROVED** [review]

**Pull request merged**